src/PerformanceTests/Common/Scenarios/BaseRunner.cs

```diff
@@ -302,4 +307,21 @@ static IEnumerable<int> IterateUntilFalse(Func<bool> condition)
         var i = 0;
         while (condition()) yield return i++;
     }
+
+    protected async Task DrainMessages()
```

Comment:
I think this method is called in the `Close()` method of every runner. If this is the case, isn't it better to add it to `Stop()` method in the `BaseRunner` until someone actually does NOT need it and another approach can be taken?